### ISSUE TITLE
Update ARIA on <va-banner> component

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#16610]
## Background
We just pulled in the `<va-banner>` component to the VBA templates that we are building. Upon review, I found some accessibility issues that should be resolved. However, in an effort to be more collaborative with the DST as well as to have our own code be more accessible quicker (because the DST has their own priorities), I would like to propose completing these updates within the Facilities team and contributing the work back to the DS using [the documented processes](https://design.va.gov/about/contributing-to-the-design-system/).

## Description
The `<va-banner>` currently has some incorrect ARIA attributes that should be adjusted to make this component more accessible: 

1. A `role="banner"` is being applied to the entire banner, however this is not correct. The banner role is intended for the global header (the `<header>` is the semantic equivalent of this role).
<img width="532" alt="Screenshot 2024-01-02 at 3 19 09 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/b46d8b5f-8823-4c54-a512-49f81f5a0bf0">

2. We can change this to `role="region"` and add `aria-label="Alert Banner"` to draw attention to this for AT.
<img width="635" alt="Screenshot 2024-01-02 at 3 50 00 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/a53ca9ae-e103-4bc1-9209-82f6503a10c1">

3. A `role="img"` has been added to the icon however the icon is also being hidden from AT, as it should be because it is decorative. Therefore we don't need the role and this should be removed.
<img width="387" alt="Screenshot 2024-01-02 at 3 34 11 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/997a8d4e-bfde-4c18-8f98-ab129495aa96">

4. A `role="presentation"` has been added to the body. Presentation is intended to remove any implicit semantics, however this particular `<div>` doesn't have any semantic accessibility and therefore this role is unnecessary and should be removed.
<img width="410" alt="Screenshot 2024-01-02 at 3 27 14 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/3c3b354c-ebfd-49c0-a788-72a208a8cffd">

## Acceptance Criteria
- [x] `role="banner"` has been changed to `role="region"`
- [x] `aria-label="Alert banner"` has been added
- [x] `role="img"` has been removed from the icon
- [x] `role="presentation"` has been removed from the body `<div>`
- [x] Confirm no testing is dependent no these roles, remove or fix if so
- [x] Accessibility review
- [x] This has been submitted to the Design System for review and approval

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [x] Facilities
- [ ] Accelerated Publishing

**Comment by laflannery** [comment]
@eselkin After looking at this, with your notes, I have some questions/comments and maybe this isn't straightforward enough for us to complete and contribute. Could we chat about it?

**Comment by eselkin** [comment]
With the close button, that `<i>` tag has a role presentation as well. Should that be removed?

**Comment by eselkin** [comment]
@laflannery Does the `aria-label` have to be on that `<div>` with class body, or could it be on the va-banner component itself?

**Comment by laflannery** [comment]
@eselkin 
1. Yes you can remove the presentation role on the close button icon. It's not necessary because we are already hiding it with `aria-hidden`
2. the `aria-label` does have to be on the same `<div>` that we are putting the `role` on. The `<va-banner>` is outside the shadow root and therefore can't be accessed by anything inside the shadow, we have to contain all of this inside or it won't work, it won't be understood.

**Comment by eselkin** [comment]
Should the `aria-label` be changeable by the user? Or should it be defined by the component type?

**Comment by laflannery** [comment]
@eselkin OH! I like that, it would be really nice if it could be defined by the component type, like "Information banner", when it's an informational alert, "Alert banner" when it's an error alert. Is that what you mean because that would be lovely

**Comment by eselkin** [comment]
@laflannery I thought I could do the automatic by component type, but because of the nesting of alert in banner it seems like I can't (or not without a complex configuration step that might make the component too slow.

**Comment by eselkin** [comment]
@laflannery Should alerts have the role="region" for that content section as well, or should we not have that.

**Comment by laflannery** [comment]
@eselkin If we add the region role to the general `<va-alert>`, that would be a case where we would want to identify `aria-labels` to each of those cases ("continue", "info", "success", "warning"), because it would be odd to have the role without a clarifying label. So I think we may want to hold on that for now since we aren't adding these individual labels.

I also want to make sure that as we do this work, we do not change the behavior of the basic `<va-alert>` in that if they use the error type, it will add `role="alert"` and we need to keep that.

**Comment by eselkin** [comment]
@laflannery right, the va-alert with status "error" still creates a role="alert"

Do you know if this causes a problem, an aria-label that is set as an attribute without a value:
`<div role="region" class="alert info" aria-label="">`

In regular va-alert the div looks like:
`<div class="alert info">`
unless it's an error status
`<div role="alert" aria-live="assertive" class="alert error">`

**Comment by eselkin** [comment]
Tested with Verdaccio (finally figured that out)
e2e Tests written
Submitted to design-system
@laflannery reviewed and approved.
Corrected issues they had with PR.

**Comment by laflannery** [comment]
I wanted to note this here for reference - There is a banner alert on the homepage now and you can see all the errors being flagged by axe that these changes are aiming to resolve:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/6ef3eb9e-54ee-47f7-8247-21b1372e4ea9)

Post-update, I can re-run axe (assuming the banner is still there) and show another screenshot with the errors not there anymore

**Comment by eselkin** [comment]
Merged in to web-components of design system

**Comment by laflannery** [comment]
@eselkin update is now on Prod, I verified and tested and everything looks good! I also re-ran axe and as expected all the previous errors that were in the screenshot above are gone:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/613b78a9-7ad2-461c-8e66-2f9a1f14b146)

**Comment by laflannery** [comment]
@xiongjaneg I'm going to close this based on the fact that this in prod now
